### PR TITLE
Navigation styling

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* Created `NavHeaderContainer` and `NavTitle` base components to standardize navigation headings and reduce repeated code. All text and buttons should be consistently positioned!
* Removed default navigation bars and added custom headings (with drop shadows) and new NavButtons from #42 on:
  * Welcome
  * Login
  * Signup
  * Store List (also added new CANCEL button consistent with Figma)
  * Products List (`ProductsScreen`)
  * Resources
  * Map Screen (transparent container)
* Cleaned up some deprecated text styles and styled-components
* Fixed a scrolling bug on the products list

### Related PRs
Inspired by the standardized NavButton base component from #42 !

## How to review
Make sure everything looks good on different screen sizes/aspect ratios.

## Next steps
_Resolved in 3/26 Update:_

- [x] I want to standardize the heading styling somehow (maybe as a base component). I'm leaving this as a draft because I need to go back and clean this up and see how I can reduce the repeated code.
- [x] I also need to fix the Rewards heading bar, but that involves fixing the tab styling.

- [x] Figure out SafeAreaView or something with `react-native-device-info` to adjust the top bar margin depending on if the device has a notch. It looks really uncomfortably big on iPhone 8 simulator right now. **Update: resolved this in #56** 

## Tests Performed, Edge Cases
* Tested on a few different screen size simulators.

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
* Styled back buttons on login and signup:
image

* Store list styled heading with new cancel button
image

* Updated Resources heading title bar
image

* No more white bar here! Map is now full screen.
image

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/map/MapScreen.js [hunk 1]
```diff
@@ -261,7 +260,7 @@ export default class MapScreen extends React.Component {
             <Subhead color={'#fff'}> Your Rewards </Subhead>
           </View>
         </TouchableOpacity>
-      </SafeAreaView>
+      </View>
```
Comment (kennethlien) [inline screens/map/MapScreen.js]: This looks so much better :D

File: screens/resources/ResourcesScreen.js [hunk 2]
```diff
@@ -45,13 +48,12 @@ class ResourcesScreen extends React.Component {
...
-
-        <View style={{ height: 106 }}>
-          <TopText>Resources</TopText>
-        </View>
+        <NavHeaderContainer>
+          <NavButton onPress={() => this.props.navigation.goBack(null)}>
+            <FontAwesome5 name="arrow-left" solid size={24} />
+          </NavButton>
+          <NavTitle>Resources</NavTitle>
+        </NavHeaderContainer>
```
Comment (tommypoa) [inline screens/resources/ResourcesScreen.js]: Looks fine on iPhone 11 but on iPhone 8 it the header/"Resources" isn't center-aligned. There is also a gap at the top of the header (also in Points History) that presents itself in < iPhone X versions - not sure if this is a problem/deviation from Figma but just flagging.
image

File: screens/rewards/RewardsScreen.js [hunk 3]
```diff
@@ -144,9 +144,9 @@ export default class RewardsScreen extends React.Component {
     return (
       <Container>
         <TopTab>
-          <BackButton onPress={() => this.props.navigation.goBack()}>
+          <NavButton onPress={() => this.props.navigation.goBack()}>
             <FontAwesome5 name="arrow-down" solid size={24} color="white" />
-          </BackButton>
+          </NavButton>
```
Comment (tommypoa) [inline screens/rewards/RewardsScreen.js]: Button isn't center-aligned / seems to be flushed to the bottom of the header, and looks like it is sticking right above the words "Healthy Rewards".
image

File: screens/auth/LoginScreen.js [hunk 4]
```diff
@@ -146,6 +148,9 @@ export default class Login extends React.Component {
   render() {
     return (
       <AuthScreenContainer>
+        <BackButton onPress={() => this.props.navigation.goBack(null)}>
+          <FontAwesome5 name="arrow-left" solid size={24} />
+        </BackButton>
```
Comment (tommypoa) [inline screens/auth/LoginScreen.js]: Is this BackButton supposed to be positioned somewhere at the top left of the screen? Fine in iPhone 11 but on iPhone 8 it appears to be closer to "Log in" than the top.
image

File: screens/resources/ResourcesScreen.js (hunk 2 above)
Comment (wangannie) [reply screens/resources/ResourcesScreen.js]: Good catch! Both of these issues were later fixed in #56

File: screens/rewards/RewardsScreen.js (hunk 3 above)
Comment (wangannie) [reply screens/rewards/RewardsScreen.js]: Interesting! It is actually properly aligned but does not appear to be because the icon has a 40px width and some padding, making the icon itself only 22px centered in the 40px so it appears misaligned. But thank you for noticing this! I will adjust it in #56

File: screens/auth/LoginScreen.js (hunk 4 above)
Comment (wangannie) [reply screens/auth/LoginScreen.js]: This is also adjusted to be higher in #56 for the keyboard covering issues. 👍
